packages/react-native/src/index.ts

```diff
@@ -0,0 +1,140 @@
+
+  checkIn(): Promise<void> {
+    throw new Error('Honeybadger.checkIn() is not yet supported on react-native')
+  }
+
+  showUserFeedbackForm(): Promise<void> {
+    throw new Error('Honeybadger.showUserFeedbackForm() is not yet supported on react-native')
+  }
+
+  private setJavascriptErrorHandler() {
+    if (this.__jsHandlerInitialized) { return }
+
+    this.logger.debug('Setting up the JavaScript global error handler.')
+    this.__originalJsHandler = ErrorUtils.getGlobalHandler();
+
+    ErrorUtils.setGlobalHandler((err, isFatal) => {
+      this.logger.debug('JavaScript global error handler triggered.')
+      this.notify(err)
+
+      // Allowing the default error handler to process the error after
+      // we're done with it will show the useful RN red info box in dev.
+      // However in prod it causes duplicate errors to be reported on iOS
+      // since the default handler re-throws a native error
+      if (__DEV__ && this.__originalJsHandler) {
+        this.logger.debug('Passing error to previous error handler.')
+        this.__originalJsHandler(err, isFatal)
+      }
+    })
+
+    this.__jsHandlerInitialized = true
+  }
+
+  private setNativeExceptionHandler() {
+    if (this.__nativeHandlerInitialized) { return }
+
+    const HoneybadgerNativeModule = NativeModules.HoneybadgerReactNative;
+    if (!HoneybadgerNativeModule) {
+      this.logger.error('The native module was not found. Please review the installation instructions.')
+      return
+    }
+  
+    HoneybadgerNativeModule.start()
+  
+    const nativeEventEmitter = new NativeEventEmitter(HoneybadgerNativeModule)
+    nativeEventEmitter.addListener(
+      'native-exception-event', 
+      this.onNativeException.bind(this)
+    )
+
+    this.__nativeHandlerInitialized = true
+  }
+
+  private onNativeException(data:NativeExceptionData) {
+    switch ( Platform.OS ) {
+    case 'ios': 
+      this.onNativeIOSException(data)
+      break
+    case 'android': 
+      this.onNativeAndroidException(data)
+      break
+    }
+  }
+
+  private onNativeIOSException(data:NativeExceptionData) {
+    const { backtrace, backtraceDetails } = backtraceAndDetailsFromIosException(data)
+    const notice = {
+      name: `React Native iOS ${data.type}`,
+      message: errorMessageFromIosException(data),
+      backtrace,
+      details: {
+        errorDomain: data.errorDomain || '',
+        initialHandler: data.initialHandler || '',
+        userInfo: data.userInfo || {},
+        architecture: data.architecture || '',
+        ...backtraceDetails
+      },
+    }
+    this.notify(notice)
+  }
+
+  private onNativeAndroidException(data:NativeExceptionData) {
+    const notice = {
+      name: `React Native Android ${data.type}`,
+      message: data.message || '',
+      details: {
+        errorDomain: data.errorDomain || '',
+        initialHandler: data.initialHandler || '',
+        userInfo: data.userInfo || {},
+        architecture: data.architecture || '',
+      },
```

Comment:
What's the difference between `details` and `context` here?
cc @joshuap 🤔

Reply:
Copied this from the original package as well -- I was trying to make as few changes as possible. I will look into it, though, I'm not sure we actually want this here.

Reply:
Thanks @joshuap!

@BethanyBerkowitz I mainly wanted to know how the `details` property is used. We don't have to change anything if this works 👌.